### DISCUSSION TITLE
Problem with custom_color_lookup on annotate function

### DISCUSSION THREAD

**Discussion by Youho99** [roboflow/supervision#1032] [Q&A] [answered]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

I'm using supervision for a project (currently I'm using supervision version 0.19)

I use the annotate() function of sv.BoundingBoxAnnotator() for example

When I use it with only the required parameters, it works correctly.

My goal is to customize the colors of the bboxes.
Actually, I'm using Streamlit, and I have one colorpicker per class created. I know how to retrieve its value (I retrieve it in hexadecimal code).
So I want to use the optional variable custom_color_lookup. However, I can't understand in what format I should pass my custom color values to it.

Let's imagine, I have 2 classes and I want to put these 2 custom colors:
custom_colors = ['#3ab63f', '#459af5']

How should I transform this variable in order to pass it without error into the optional custom_color_lookup variable of the annotate() function?

Thanks for your help

**Comment by SkalskiP** [comment]
Hi, let me convert this issue into a discussion and move it to the Q&A section.

**Comment by SkalskiP** [comment, marked as answer]
Hi @Youho99 👋🏻, `custom_color_lookup` is not the feature you are looking for. You should use [`sv.ColorPalette`](https://supervision.roboflow.com/latest/draw/color/#supervision.draw.color.ColorPalette.from_hex). Now you will be able to plug it into any Supervision annotator.

```python
import supervision as sv

color = sv.ColorPalette.from_hex(['#3ab63f', '#459af5'])

annotator = sv.BoundingBoxAnnotator(color=color)
```

**Reply by Ying-Kang** [reply]
this strategy can only change the text background color

what should i do if i want to unify the box color into the same?

**Reply by SkalskiP** [reply]
Hi @Ying-Kang 👋🏻 just pass the same ColorPalette to all annotators you use.
